src/tree/README.md

```diff
@@ -0,0 +1,285 @@
+            "Title":null,
+            "Description":null,
+            "Operations":[
+               
+            ]
+         }
+      }
+   ],
+   "FirstPublished":{
+      "Date":"2021-07-06T07:12:02.153Z",
+      "User":"Admin Admin",
+      "Id":null
+   },
+   "LastPublished":{
+      "Date":"2021-07-06T07:12:02.153Z",
+      "User":"Admin Admin",
+      "Id":null
+   },
+   "RelatedEvents":[
+      
+   ],
+   "RelatedNews":[
+      
+   ],
+   "SimilarNews":[
+      
+   ]
+}
+```
+
+> Please note that the search functionality in the related data fields (and in the entire AdminApp) is based on the item's primary identifier, which in the case of the default content modules is the `Title`. When you modify the component rendered in the tree's item template, it will not affect the search feature. If you would like to add another field to be included in the search, please see the [search config]("./../../search").
+
+## Development process
+
+### Implement the component
+
+Implement a component as you normally would in Angular, with just one requirement, that your custom component **must** extend the `abstract class CustomTreeComponentBase`, it will force you to implement the only property `item: any`, which as stated on top, will be populated by the item that is going to be rendered.
+
+```typescript
+import { Component, Input } from "@angular/core";
+import { CustomTreeComponentBase } from "@progress/sitefinity-adminapp-sdk/app/api/v1/tree";
+
+@Component({
+    template: `
+    <div>{{item?.Title}} - {{item?.CreatedBy}}</div>
+    <div class="sf-tree__description sf-breadcrumb -sf-overflow-ellipsis">
+        <span>
+            Created on {{item?.DateCreated}}
+        </span>
+    </div>
+    `
+})
+export class RelatedDataCustomComponent extends CustomTreeComponentBase {
+    @Input() item: any;
+}
+
+```
+
+### Registering the component
+
+The component you have implemented must be registered in a class that implements the `TreeNodeComponentProvider` interface. The interface has a single method `getComponentData` which accepts two parameters, `feature` of type `TreeNodeComponentFeatures`, which currently supports only one feature - the related data, and `entitySet` which is a `string`, intended to represent the currently loaded data. This method will be called by the AdminApp, for example if we assume that your project's news items, have a related data field for events, the values passed in the method will `TreeNodeComponentFeatures.RelatedData` and `events`, therefore your implementation of the provider should return a `ComponentData` object that you would like to be used when displaying the related data nodes in the content item.
+
+* "albums"
+* "blogposts"
+* "blogs"
+* "calendars"
+* "contentitems"
+* "documentlibraries"
+* "events"
+* "flat-taxa"
+* "folders"
+* "form-drafts"
+* "hierarchy-taxa"
+* "images"
+* "listitems"
+* "lists"
+* "newsitems"
+* "pages"
+* "sites"
+* "taxonomies"
+* "videolibraries"
+* "videos"
```

Comment:
Do we really need this list?

Reply:
I believe they already know how to get the entity set name and these explanations are obsolete

Reply:
removing

Reply:
removed